cupyx/scipy/stats/_stats_py.py

```diff
@@ -0,0 +1,147 @@
+import warnings
+
+import cupy
+
+
+def _contains_nan(a, nan_policy='propagate'):
+    policies = ['propagate', 'raise', 'omit']
+
+    if nan_policy not in policies:
+        raise ValueError("nan_policy must be one of {%s}" %
+                         ', '.join("'%s'" % s for s in policies))
+
+    try:
+        contains_nan = cupy.isnan(cupy.sum(a))
+    except TypeError:
+        try:
+            contains_nan = cupy.nan in set(a.ravel())
```

Comment:
We cannot pass cupy.ndarray to make a set. When do you think the line above `contains_nan = cupy.isnan(cupy.sum(a))` raises a TypeError, and this `except` block is needed?

Reply:
SciPy says: [Link here](https://github.com/scipy/scipy/blob/main/scipy/stats/_stats_py.py#L108-L111)
```txt
# This can happen when attempting to sum things which are not
# numbers (e.g. as in the function `mode`). Try an alternative method:
```

Shall I remove it?

Reply:
Thanks, let's remove it!